src/callbacks.jl

```diff
@@ -39,78 +42,110 @@ function addInfoCallback(m::Model, f::Function)
     push!(m.callbacks, InfoCallback(f))
 end
 
-function attach_callbacks(m::Model, cbs::Vector{LazyCallback})
+function lazycallback(d::MathProgBase.MathProgCallbackData, m::Model, cbs::Vector{LazyCallback})
+    state = MathProgBase.cbgetstate(d)
+    @assert state == :MIPSol || state == :MIPNode
     anyfrac = mapreduce(|, cbs) do cb
         cb.fractional
     end
-    function lazycallback(d::MathProgBase.MathProgCallbackData)
-        state = MathProgBase.cbgetstate(d)
-        @assert state == :MIPSol || state == :MIPNode
-        if state == :MIPSol
-            MathProgBase.cbgetmipsolution(d,m.colVal)
-        elseif anyfrac
-            MathProgBase.cbgetlpsolution(d,m.colVal)
-        else
-            return nothing
-        end
+    if state == :MIPSol
+        MathProgBase.cbgetmipsolution(d,m.colVal)
+    elseif anyfrac
+        MathProgBase.cbgetlpsolution(d,m.colVal)
+    else
+        return :Continue
+    end
+    try
         for cb in cbs
             if state == :MIPSol || cb.fractional
                 cb.f(d)
             end
         end
-        nothing
+    catch y
+        if isa(y, CallbackAbort)
+            return :Exit
+        else
+            rethrow(y)
+        end
     end
-    MathProgBase.setlazycallback!(m.internalModel, lazycallback)
+    :Continue
 end
 
-function attach_callbacks(m::Model, cbs::Vector{CutCallback})
-    function cutcallback(d::MathProgBase.MathProgCallbackData)
-        state = MathProgBase.cbgetstate(d)
-        @assert state == :MIPSol || state == :MIPNode
-        if state == :MIPSol  # This shouldn't happen right?
-            println("Is this ever called?")
-            MathProgBase.cbgetmipsolution(d,m.colVal)
-        else
-            MathProgBase.cbgetlpsolution(d,m.colVal)
-        end
+attach_callbacks(m::Model, cbs::Vector{LazyCallback}) =
+    MathProgBase.setlazycallback!(m.internalModel, d -> lazycallback(d,m,cbs))
+
+function cutcallback(d::MathProgBase.MathProgCallbackData, m::Model, cbs::Vector{CutCallback})
+    state = MathProgBase.cbgetstate(d)
+    @assert state == :MIPSol || state == :MIPNode
+    if state == :MIPSol  # This shouldn't happen right?
```

Comment:
That would mean the ``state == :MIPNode`` branch becomes an error, but I guess that'll never really be called anyway.